doc/lua_api.txt

```diff
@@ -6861,6 +6861,8 @@ and [`PlayerMetaRef`].
 * `get_int(key)`: Returns `0` if key not present.
 * `set_float(key, value)`
 * `get_float(key)`: Returns `0` if key not present.
+* `set_bool(key, value)`
+* `get_bool(key)`: Returns `false` if key not present.
```

Comment:
Maybe this should have an optional `default` argument like `Settings:get_bool`. `meta:get_bool("...") or true` won't work as expected.

Reply:
That wouldn't be consistent with all the other `meta:get_*` functions, which suffer the same issue. Should I add it anyway?

Reply:
It would be more consistent with the settings API's get_bool

Reply:
What about adding a `default` argument to _all_ `meta:get_ ` functions in a separate commit?

Reply:
> That wouldn't be consistent with all the other `meta:get_*` functions, which suffer the same issue. Should I add it anyway?

It doesn't have to be. This is a special issue with booleans because `false` is falsey, thus the commonly used `x or y` construction will fail as TurkeyMcMac has pointed out already. It is a non-issue for strings or numbers (as long as the functions return `nil` for absent values).

Reply:
I suppose it should be noted that `get_int` etc. do not return `nil` for an absent key, but in their case one can just write `tonumber(meta:get("key")) or default`.

Reply:
Doesn't the same hold for booleans? For example:

```lua
meta:get("key") == nil or meta:get("key") == "true"
```

Reply:
I suppose one could write `minetest.is_yes(meta:get(key) or "true")`. So actually you're right, the default argument is not needed.

Reply:
The cleanest & most efficient for modders would probably be writing a utility `local function default(x, def) return x == nil and def or x end`, then use that.

Reply:
I'm late, but wouldn't be `nil` as default be preferable? (I.e. for `get_bool(key) ~= false`.)

Also, it might be worth documenting that the stored string is correctly interpreted by `minetest.is_yes`.